dagshub/data_engine/client/data_client.py

```diff
@@ -180,8 +184,27 @@ def _datasource_query(
             first=limit,
             after=after,
         )
-
-        return self._exec(q, params)["datasourceQuery"]
+        q.validate_params(params, self.introspection_dict)
+        return self._exec(q.generate(), params)["datasourceQuery"]
+
+    @cached_property
+    def introspection_dict(self):
+        introspection = """
+{
+  __schema {
+    types {
+      name
+      inputFields {
+        name
+        type {
+          name
+        }
+      }
+    }
+  }
+}
+        """
+        return self._exec(introspection)
```

Comment:
I know that this is a "do it right now" thing, but I want it to be done okay-ishly so we're not stuck with filtering through dict.

Make the result of this into a Dacite model. That way we can have the validators on that model object.